src/core/ddsi/include/dds/ddsi/shm_types.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright(c) 2021 Apex.AI Inc. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+#ifndef SHM_TYPES_H
+#define SHM_TYPES_H
+
+#include "dds/ddsi/q_protocol.h" /* for, e.g., SubmessageKind_t */
+#include "dds/ddsi/ddsi_tran.h"
+#include "dds/features.h"
+
+#include "dds/ddsi/ddsi_keyhash.h"
+#include "iceoryx_binding_c/chunk.h"
+
```

Comment:
I think this header is also missing an ` extern “C”` block for C linkage, since this will also be used in the C++ code